services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDelegateMode.java

```diff
@@ -349,41 +363,130 @@ public <H, P extends ResourcePath<K>, K, R> Scatter<H, P, K> scatter(
     return finalScatter;
   }
 
-  // Select host with the least pending queue depth.
-  private <H> H selectLeastLoadedHost(List<H> hosts, VenicePath path) throws RouterException {
-    H host;
-    long minCount = Long.MAX_VALUE;
+  /**
+   * Select the least loaded host from the available healthy replicas.
+   * Behavior depends on the latencyBasedRoutingEnabled config flag:
+   *
+   * When enabled, uses average response latency as the criterion with a 1.5x spectrum
+   * threshold among the healthy host list. Ref {@link #AVG_LATENCY_SPECTRUM_FOR_HOST_SELECTION}.
+   * Randomly selects within the latency spectrum to avoid skewed traffic.
+   *
+   * When disabled (default), uses pending request count to decide on the host selection
+   * among the healthy host list.
+   *
+   * Note: Healthy host list is based on pending queue count: Ref {@link VeniceHostHealth#isHostHealthy}
+   */
+  private <H> H selectLeastLoadedHost(List<H> hosts, VenicePath path, RouteHttpStats routeHttpStats)
+      throws RouterException {
+    if (latencyBasedRoutingEnabled) {
+      return selectLeastLoadedHostByLatency(hosts, path, routeHttpStats);
+    } else {
+      return selectLeastLoadedHostByPendingCount(hosts, path);
+    }
+  }
+
+  /**
+   * Select host with the lowest pending request count:
+   * This has concerns where a host with low pending count but high latency could be
+   * selected when the QPS is low, leading to suboptimal routing decisions.
+   */
+  private <H> H selectLeastLoadedHostByPendingCount(List<H> hosts, VenicePath path) throws RouterException {
     H minHost = null;
+    long smallestPendingCount = Long.MAX_VALUE;
+
     for (H h: hosts) {
```

Comment:
In FC implementation we actually shuffle the hosts every time so we don't always route to the same replica when QPS is relatively low and servers are fast (pending request count is all the same which is at 0). Should we do that here too?